### PR TITLE
type: enable type prompt for .dumirc.ts

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

```
Please append `.dumi/**/*` & `.dumirc.ts` into `include` option of `tsconfig.json`, to make sure the types exported by framework works.
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | --          |
| 🇨🇳 Chinese |  --         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d65ab80</samp>

Enabled TypeScript support for the documentation site by modifying the `tsconfig.json` file for dumi. Added the `.dumirc.ts` file to the `include` property and extended the base `tsconfig.json` file.

### <samp>🤖 Generated by Copilot at d65ab80</samp>

* Enable TypeScript support for the documentation site by:
  - Including the `.dumirc.ts` file in the `include` property of `.dumi/tsconfig.json` ([link](https://github.com/ant-design/ant-design/pull/44886/files?diff=unified&w=0#diff-1dc9271d36bc7b7434564ed89afe8bdd79f74f22ee03d5482e82ee0be7c600a1L14-R15))
  - Extending the base `tsconfig.json` file from the root directory in the `extends` property of `.dumi/tsconfig.json` ([link](https://github.com/ant-design/ant-design/pull/44886/files?diff=unified&w=0#diff-1dc9271d36bc7b7434564ed89afe8bdd79f74f22ee03d5482e82ee0be7c600a1L14-R15))

### CODE REVIEW

**Comment by afc163** [comment]
.dumi 里这么多报错。。。

**Comment by li-jia-nan** [comment]
> .dumi 里这么多报错。。。

我来修

**Comment by li-jia-nan** [comment]
本地居然不报错，奇怪

**Comment by MadCcc** [comment]
.dumi 文件夹有自己的 tsconfig

File: .dumi/tsconfig.json [hunk 1]
```diff
@@ -11,5 +11,6 @@
       "dumi/theme/*": [".dumi/theme/*"]
     }
   },
-  "include": ["./**/*", "../site/theme/template/Content"]
+   "include": ["./**/*", "../.dumirc.ts"],
+   "extends": "../tsconfig.json"
```
Comment (MadCcc) [inline .dumi/tsconfig.json:15]: extends 写最上面吧

**Comment by PeachScript** [comment]
dumi 层的 warning 有问题，`.dumi/**` 不应该包含进来，这个 PR 我来跟进吧，先把 dumi 层的 warning 修了再看 PR 里怎么调整

File: .dumi/tsconfig.json (hunk 1 above)
Comment (li-jia-nan) [reply .dumi/tsconfig.json]: done~

**Comment by PeachScript** [comment]
已更新，大家再 review 看看

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**